backend/ai_firm/report_generation.py

```
@@ -930,12 +966,28 @@ def _section_11_trade_setups(self, symbol, data):
+        try:
+            from services.market_data_service_waterfall import get_waterfall_service
+            wf = get_waterfall_service()
+            logs = wf.get_recent_audit_logs(1)
+            if logs and logs[0].get('audit_id') == audit_id:
+                log = logs[0]
+                return f"""### 12. AUDIT LOG (LIVE)
+
+| Section | Data Age | Source(s) | Fallback | ID | Status |
+|---|---|---|---|---|---|
+| Price/Verified | {log.get('timestamp')} | {', '.join(log.get('sources_used', []))} | Level {log.get('fallback_level')} | {log.get('audit_id')} | {log.get('verification_status')} |"""
+        except:
```

Flake8 (7.3.0)

[error] do not use bare 'except'

(E722)